error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/NestedPublishers.java

```diff
@@ -0,0 +1,74 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.LinkType.CUSTOM;
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+import static com.google.errorprone.BugPattern.StandardTags.FRAGILE_CODE;
+import static tech.picnic.errorprone.bugpatterns.util.Documentation.BUG_PATTERNS_BASE_URL;
+import static tech.picnic.errorprone.bugpatterns.util.MoreTypes.generic;
+import static tech.picnic.errorprone.bugpatterns.util.MoreTypes.subOf;
+import static tech.picnic.errorprone.bugpatterns.util.MoreTypes.type;
+import static tech.picnic.errorprone.bugpatterns.util.MoreTypes.unbound;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.suppliers.Suppliers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Type;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+/**
+ * A {@link BugChecker} that flags nesting of {@link Publisher Publishers} inside {@link Mono
+ * Monos}.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+    summary =
+        "Avoid nesting `Publisher`s inside `Publishers`s; the resultant code is hard to reason about",
+    link = BUG_PATTERNS_BASE_URL + "NestedPublishers",
+    linkType = CUSTOM,
+    severity = WARNING,
+    tags = FRAGILE_CODE)
+public final class NestedPublishers extends BugChecker implements MethodInvocationTreeMatcher {
+  private static final long serialVersionUID = 1L;
+  private static final Supplier<Type> PUBLISHER = type("org.reactivestreams.Publisher");
+  private static final Supplier<Type> PUBLISHER_OF_PUBLISHERS =
+      VisitorState.memoize(
+          generic(PUBLISHER, subOf(generic(type("org.reactivestreams.Publisher"), unbound()))));
+  private static final Supplier<Type> GROUPED_FLUX =
```

Comment:
Should we extend this to all classes that extend either a `Mono` or `Flux` ?

Reference for question: 
```
public abstract class GroupedFlux<K, V> extends Flux<V> {
```

Reply:
not sure I understand the question; you're matching `GroupedFlux` with any `<K,V>` right (unbounded)?

Reply:
Sorry, I re-read my question and realised it is not clear 😓 .

I meant, we are using Type `GROUPED_FLUX` to exclude something like this `Flux<GroupedFlux<K,V>>` in the `matchMethodInvocation` method.

Since `GroupedFlux` extends `Flux`, I was asking if we can make `GROUPED_FLUX` type more generic and to match on something anything extending either a flux or a mono like `? extends Flux` ( See also `ParallelFlux`, `ConnectableFlux`). 

Do I make more sense now ?
It also could be there is something I am missing 😅

Reply:
Discussed offline: Let's exclude the only know use case for now. 👌